_search-plugins/search-pipelines/rag-processor.md

```diff
@@ -0,0 +1,97 @@
+
+## Request fields
+
+The following table lists all available request fields.
+
+Field | Data type | Description
+:--- | :--- | :---
+`model_id` | String | The ID of the model used in the pipeline. Required.
+`context_field_list` | Array | A list of fields in document sources that the pipeline uses as context for RAG. Required. For more information, see [Context field list](#context-field-list). 
+`system_prompt` | String | The system prompt that is sent to the LLM to help adjust its behavior, such as response tone. Can be a persona description or a set of instructions. Optional.
+`user_instructions` | String | Human-generated instructions sent to the LLM to guide it in producing the results. 
+`tag` | String | The processor's identifier. Optional.
+`description` | String | A description of the processor. Optional.
+
+### Context field list
+
+The `context_field_list` is a list of fields in document sources that the pipeline uses as context for RAG. For example, let's say your OpenSearch index stores a collection of documents, each containing a `title` and `text`:
+
+```json
+{
+  "_index": "qa_demo",
+  "_id": "SimKcIoBOVKVCYpk1IL-",
+  "_source": {
+    "title": "Abraham Lincoln 2",
+    "text": "Abraham Lincoln was born on February 12, 1809, the second child of Thomas Lincoln and Nancy Hanks Lincoln, in a log cabin on Sinking Spring Farm near Hodgenville, Kentucky.[2] He was a descendant of Samuel Lincoln, an Englishman who migrated from Hingham, Norfolk, to its namesake, Hingham, Massachusetts, in 1638. The family then migrated west, passing through New Jersey, Pennsylvania, and Virginia.[3] Lincoln was also a descendant of the Harrison family of Virginia; his paternal grandfather and namesake, Captain Abraham Lincoln and wife Bathsheba (née Herring) moved the family from Virginia to Jefferson County, Kentucky.[b] The captain was killed in an Indian raid in 1786.[5] His children, including eight-year-old Thomas, Abraham's father, witnessed the attack.[6][c] Thomas then worked at odd jobs in Kentucky and Tennessee before the family settled in Hardin County, Kentucky, in the early 1800s.[6]\n"
+  }
+}
+```
+
+You can specify to send only the `text` contents to the LLM by setting `"context_field_list": ["text"]` in the processor. 
+
+## Example 
+
+The following example demonstrates using a search pipeline with a `retrieval_augmented_generation` processor. 
+
+### Creating a search pipeline 
+
+The following request creates a search pipeline containing a `retrieval_augmented_generation` processor for an OpenAI model:
+
+```json
+PUT /_search/pipeline/rag_pipeline
+{
+  "response_processors": [
+    {
+      "retrieval_augmented_generation": {
```

Comment:
For this rag-processor, there is limitation to it, and we should specify in the document. @ylwu-amzn  AFAIK, is it currently the `retrieval_augmented_generation` processor only support OPENAI model?

Reply:
Added limitations.